sql/moz-fx-data-shared-prod/firefox_ios_derived/app_store_retention_week_4_v1/query.sql

```diff
@@ -0,0 +1,52 @@
+WITH clients_retention AS (
+  SELECT
+    submission_date,
+    client_id,
+    sample_id,
+    days_seen_bits,
+  FROM
+    firefox_ios.baseline_clients_last_seen
+  WHERE
+    submission_date = @submission_date
+    AND normalized_channel = "release"
+),
+first_seen AS (
+  SELECT
+    client_id,
+    sample_id,
+    first_seen_date,
+  FROM
+    firefox_ios.baseline_clients_first_seen
+  -- 28 days need to elapse before calculating the week 4 and day 28 retention metrics
+  WHERE
+    submission_date = DATE_SUB(@submission_date, INTERVAL 27 DAY)
+),
+retention_calculation AS (
+  SELECT
+    first_seen.first_seen_date,
+    clients_retention.client_id,
+    clients_retention.sample_id,
+    LENGTH(REPLACE(mozfun.bits28.to_string(days_seen_bits), "0", "")) AS days_seen_in_first_28_days,
+    mozfun.bits28.retention(days_seen_bits, @submission_date) AS retention,
+  FROM
+    clients_retention
+  INNER JOIN
+    first_seen
+  USING
+    (client_id, sample_id)
+)
+SELECT
+  * EXCEPT (retention) REPLACE(
+    -- metric date should align with first_seen_date, if that is not the case then the query will fail.
+    IF(
+      retention.day_27.metric_date <> first_seen_date,
+      ERROR("Metric date misaligned with first_seen_date"),
+      first_seen_date
+    ) AS first_seen_date
+  ),
+  days_seen_in_first_28_days > 1 AS repeat_first_month_user,
+  -- added for testing.
+  retention.day_27.active_in_week_1 AS retained_week_2,
```

Comment:
This one is not correct. What we need here is: "For users who started 27 days ago, which were active in days 7-27 (0-indexed)"? Otherwise, we run the risk of a non-consecutive funnel: week-4 retained users who were not week-2 retained.

Note that with this change, the value will be different than the week 2 retention in the previous table.

Reply:
cc @bani @soGaussian if we want to include other funnel metrics, we need to outline them in the decision brief

Reply:
@fbertsch not sure I follow your first comment, are you referring to repeat first month user metric or retained week 2 metric. If it is the repeat first month user metric, then the logic is correct, the way we defined it in the brief was any new profile that returned again with the first 27 days after their first day. The retained week 2 metric is not part of the funnel, it is a metric @kik-kik  add because the UDF calculated that along with the other retention metrics.